Mathlib/Algebra/Category/ModuleCat/Differentials/Presheaf.lean

```diff
@@ -0,0 +1,231 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Category.ModuleCat.Presheaf
+import Mathlib.Algebra.Category.ModuleCat.Differentials.Basic
+
+/-!
+# The presheaf of differentials of a presheaf of modules
+
+In this file, we define the type `M.Derivation φ` of derivations
+with values in a presheaf of `R`-modules `M` relative to
+a morphism of `φ : S ⟶ F.op ⋙ R` of presheaves of commutative rings
+over categories `C` and `D` that are related by a functor `F : C ⥤ D`.
+We formalize the notion of universal derivation.
```

Comment:
Trying to read this first paragraph, I found myself not knowing what S was, and actually I'm still confused mathematically. If F : C => D then F.op >>> R is a presheaf of rings on D,and I'm guessing that S is a scheme because that's what it is in the next paragraph, and now I don't know what kind of object phi is supposed to be. Please feel free to add explanations so that I understand the first paragraph :-)